File: proselint/checks/misc/greylist.py
"""
Use of greylisted words.

---
layout:     post
source:     Strunk & White
source_url: ???
title:      Use of greylisted words
date:       2014-06-10 12:31:19
categories: writing
---

Use of greylisted words.

"""

from itertools import chain

from proselint.registry.checks import Check, CheckResult, types

GREYLISTED_TERMS = {
    "obviously": "This is obviously an inadvisable word to use.",
    "utilize": "Do you know anyone who needs to utilize the word utilize?",
}


def _check_greylist(text: str, check: Check) -> list[CheckResult]:
    """Check the text."""
    return list(
        chain.from_iterable(
            types.ExistenceSimple(pattern=term).check(
                text,
                Check(
                    check_type=check.check_type,
                    path=check.path,
                    message=check.message.format(term, explanation),
                ),
            )
            for term, explanation in GREYLISTED_TERMS.items()
        )
    )


check_greylist = Check(
    check_type=_check_greylist,
    path="misc.greylist",
    message="Use of '{}'. {}",
)

__register__ = (check_greylist,)

File: proselint-rs/crates/proselint-checks/src/misc/greylist.rs
use proselint_registry::{pad, checks::{Check, CheckType, Padding}};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &[
	"She should utilize her knowledge.",
	"This is obviously an inadvisable word to use obviously.",
	"I utilize a hammer to drive nails into wood.",
	"Do you know anyone who *needs* to utilize the word utilize?",
];

const CHECK_OBVIOUSLY: Check = Check {
	check_type: CheckType::ExistenceSimple {
		pattern: pad!(Padding::WordsInText, "obviously"),
		exceptions: &[],
	},
	path: "misc.greylist.obviously",
	msg: "Use of 'obviously'. This is obviously an inadvisable word to use.",
	..Check::default()
};

const CHECK_UTILIZE: Check = Check {
	check_type: CheckType::ExistenceSimple {
		pattern: pad!(Padding::WordsInText, "utilize"),
		exceptions: &[],
	},
	path: "misc.greylist.utilize",
	msg: "Use of 'utilize'. \
	Do you know anyone who needs to utilize the word utilize?",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK_OBVIOUSLY, CHECK_UTILIZE];
